### ISSUE TITLE
Inline editing - performance issues

### ISSUE THREAD

**Issue by andreme** [GriddleGriddle/Griddle#728]
I was looking at using Immutable records to improve performance and while digging into it I was wondering why the rowDataSelector uses a find instead of the lookup array?

Original
```javascript
export const rowDataSelector = (state, { griddleKey }) => {
  return state.get('data')
    .find(r => r.get('griddleKey') === griddleKey).toJSON();
};
```
Updated
```javascript
export const rowDataSelector = (state, { griddleKey }) => {
  return state.get('data').get(state.getIn(['lookup', griddleKey.toString()])).toJSON();
};
```

**Comment by andreme** [comment]
I'm also fairly sure that there is a bug in GRIDDLE_UPDATE_STATE,

 ```javascript
const newState = _.omit(action.newState, data);
```
should be
 ```javascript
const newState = _.omit(action.newState, 'data');
```
?

**Comment by ryanlanciaux** [comment]
Thanks a ton for looking into this! Everything should be using the lookup where possible so maybe that was something that was missed in the transition. 

There very well could be a bug with the _.omit and update state.

**Comment by andreme** [comment]
I've fixed these things in #729

**Comment by andreme** [comment]
One reason that Rows and Cells get re-rendered is the use of Immutable.toJSON() in rowDataSelector, rowPropertiesSelector, cellPropertiesSelector.
.toJSON() generates a new JS object every time it's called, breaking reacts shallow compare for props.
